### PR TITLE
feat(translation): add functionality to enable the use of translations in Stark modules

### PR DESCRIPTION
**Pull request by RobbyDeLaet**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

All translations are defined in the app.

Issue Number: #53

## What is the new behavior?
Each module can have it's own translations.
There are 3 locations where translations can be defined:
 - Common (shared by more than one module)
 - Module
 - App
In case of duplicate keys, the hierarchy is: common -> module -> app
Meaning that the app has always the upper hand.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: packages/stark-core/src/common/translations/translations.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { TranslateService } from "@ngx-translate/core";
+const _cloneDeep: Function = require("lodash/cloneDeep");
+
+import { StarkLocale } from "./locale.intf";
+import { translationsEn } from "./translations/en";
+import { translationsFr } from "./translations/fr";
+import { translationsNl } from "./translations/nl";
+
+export function mergeTranslations(translateService: TranslateService, ...args: StarkLocale[]): void {
```
Comment (christophercr) [inline packages/stark-core/src/common/translations/translations.ts]: Could you maybe add a JSDoc for this function to describe what it actually does and why?

File: packages/stark-core/src/common/translations/translations/en.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export const translationsEn: any = {
```
Comment (christophercr) [inline packages/stark-core/src/common/translations/translations/en.ts]: You can replace the type "any" by "object" so at least it is more specific...

File: starter/src/app/app.module.ts [hunk 3]
```diff
@@ -25,6 +24,15 @@ import {
 } from "@nationalbankbelgium/stark-core";
 import { routerConfigFn } from "./router.config";
 import { Deserialize } from "cerialize";
+
+/*
+ * Translations
+ */
+import { TranslateModule, TranslateService } from "@ngx-translate/core";
+const translationsEn: any = require("../../assets/translations/en.json");
```
Comment (christophercr) [inline starter/src/app/app.module.ts]: You can replace the type "any" by "object" so at least it is more specific...

File: starter/assets/translations/fr.json [hunk 4]
```diff
@@ -1,170 +1,220 @@
...
+
+    "PRELOADING": {
+      "FETCHING_USER_PROFILE": "Initialisation...",
+      "FETCHING_USER_PROFILE_FAILURE": "Échec de l'initialisation: impossible de récupérer le profil de l'utilisateur.",
+      "CONTACT_IT_SUPPORT": "Veuillez contacter votre service d'assistance informatique.",
+      "CORRELATION_ID": "ID de corrélation",
+      "RELOAD": "Recharger"
+    },
+
+    "RBAC": {
```
Comment (dsebastien) [inline starter/assets/translations/fr.json]: The RBAC translations will have to be moved later on to their own module.

File: .release-it.json [hunk 5]
```diff
@@ -5,8 +5,8 @@
 	"force": false,
 	"pkgFiles": ["package.json"],
 	"increment": "conventional:angular",
-  "beforeChangelogCommand": "npm run generate:changelog",
```
Comment (dsebastien) [inline .release-it.json]: Same comment as for Christopher’s latest PR, I think we should try to isolate styling changes in their own ‘chore’ PRa to limit the noise.

File: starter/assets/translations/fr.json (hunk 4 above)
Comment (RobbyDeLaet) [reply starter/assets/translations/fr.json]: All translation need to be moved to their corresponding modules when we create them in stark-ui.

**Pull request merged**